packages/docs/src/components/DocSiteNavigation.tsx

```diff
@@ -112,69 +86,10 @@ const DocSiteNavigation = ({ location }: DocSiteNavProps) => {
     }
   `);
 
-  /**
-   * determines if the item name is included in the location pathname
-   */
-  const isItemSelected = (name: string) => {
-    return location?.pathname.includes(name);
-  };
-
-  /**
-   * Checks sub nav items to see if any are currently selected
-   */
-  const isSubNavItemSelected = (subNavItems) => {
-    return subNavItems.some((navItem) => isItemSelected(navItem.url));
-  };
-
-  /**
-   * Updating a name to remove kebab case & get rid of numeric ordering
-   */
-  const formatNavItemLabel = (name: string): string => {
-    let newName = name.replace(/-/g, ' ');
-    newName = removePositioning(newName);
-    return newName;
-  };
-
-  const formatNavItemData = ({ name, relativePath }: NavItem) => {
-    const url = makePageUrl(relativePath);
-    return {
-      label: formatNavItemLabel(name),
-      url,
-      id: url,
-      selected: isItemSelected(url),
-    };
-  };
-
-  const formatNavData = (dataList: GraphQlNavItem[]): VerticalNavItemProps[] => {
-    const retVal: VerticalNavItemProps[] = [];
-    dataList.forEach((dataItem) => {
-      // for each level 1 item that has sub nav items
-      if (dataItem.fieldValue.length) {
-        // format all the level 2 items
-        const subNavItems = dataItem.edges.map((subNavItem) => formatNavItemData(subNavItem.node));
-
-        const labelText = formatNavItemLabel(dataItem.fieldValue);
-        const isSelected = isItemSelected(dataItem.fieldValue);
-        // add level 1 item & sub items
-        retVal.push({
-          label: labelText,
-          items: subNavItems,
-          defaultCollapsed: !isSubNavItemSelected(subNavItems),
-          selected: isSelected,
-        });
-      } else {
-        // for each level 1 item without sub nav items,
-        // add each top level item
-        dataItem.edges.forEach((navItemLvl1) => {
-          retVal.push(formatNavItemData(navItemLvl1.node));
-        });
-      }
-    });
-
-    return retVal;
-  };
-
-  const navItems: VerticalNavItemProps[] = formatNavData(data?.allFile?.group);
+  const navItems: VerticalNavItemProps[] = useMemo(() => {
```

Comment:
Wanted to call out this fun use of `useMemo`. I did some logging and it seems like the data restructuring was happening on certain re-renders, so by using this memoization, the data restructuring only happens if the data or the window location data changes

Reply:
Nice! 🙌